templates/pages/setup/general/general_setup.html.twig

```diff
@@ -0,0 +1,337 @@
+        __("Don't show search engine in dropdowns if the number of items is less than"),
+        inline_field_options|merge({
+            'min': 1,
+            'max': 200,
+            'toadd': [__('Never')],
+            'width': 'auto',
+        })
+    ) }}
+</div>
+
+<div class="hr-text">
+    <i class="ti ti-search"></i>
+    <span>{{ __('Search engine') }}</span>
+</div>
+
+<div class="row ps-4">
+
+    {{ fields.dropdownArrayField(
+        'allow_search_view',
+        config['allow_search_view'],
+        [
+            __('No'),
+            __('%1$s (%2$s)')|format(__('Yes'), __('last criterion')),
+            __('%1$s (%2$s)')|format(__('Yes'), __('default criterion'))
+        ],
+        __('Items seen'),
+        inline_field_options|merge({
+            'width': 'auto',
+        })
+    ) }}
+
+    {% set allow_search_view_options = inline_field_options %}
+    {% if not config['allow_search_view'] %}
+        {% set allow_search_view_options = allow_search_view_options|merge({
+            'readonly': true,
+            'disabled': true,
+        }) %}
+    {% endif %}
+
+    {{ fields.dropdownYesNo(
+        'allow_search_global',
+        config['allow_search_global'],
+        __('Global search'),
+        allow_search_view_options|merge({
+            'width': 'auto',
+        })
+    ) }}
+
+    {{ fields.dropdownArrayField(
+        'allow_search_all',
+        config['allow_search_all'],
+        [
+            __('No'),
+            __('%1$s (%2$s)')|format(__('Yes'), __('last criterion')),
+        ],
+        __('All'),
+        inline_field_options|merge({
+            'width': 'auto',
+        })
+    ) }}
+
+    {{ fields.numberField(
+        'list_limit_max',
+        config['list_limit_max'],
+        __('Default search results limit (page)'),
+        inline_field_options|merge({
+            'min': 5,
+            'max': 200,
+            'step': 5,
+        })
+    ) }}
+
+    {{ fields.numberField(
+        'cut',
+        config['cut'],
+        __('Default characters limit (summary text boxes)'),
+        inline_field_options|merge({
+            'step': 50,
+        })
+    ) }}
+
+    {{ fields.numberField(
+        'url_maxlength',
+        config['url_maxlength'],
+        __('Default url length limit'),
+        inline_field_options|merge({
+            'min': 20,
+            'max': 80,
+            'step': 5,
+        })
+    ) }}
+
+</div>
+
+<div class="hr-text">
+    <i class="ti ti-lock"></i>
+    <span>{{ __('Item locks') }}</span>
+</div>
+
+<div class="row ps-4">
+
+    {{ fields.checkboxField(
+        'lock_use_lock_item',
+        config['lock_use_lock_item'],
+        __('Use locks'),
+        inline_field_options
+    ) }}
+
+    {% set lock_options = field_options %}
+    {% if not config['lock_use_lock_item'] %}
+        {% set lock_options = lock_options|merge({
+            'readonly': true,
+            'disabled': true,
+        }) %}
+    {% endif %}
```

Comment:
It forces user to first set 'Yes' , then save to be able to configure the feature. Maybe a small javascript listener could enhance UX.